kimchi/src/circuits/domains.rs

```diff
@@ -21,24 +23,33 @@ impl<F: FftField> EvaluationDomains<F> {
     /// and `d8` (of size `8n`). If generator of `d8` is `g`, the generator
     /// of `d4` is `g^2`, the generator of `d2` is `g^4`, and the generator of `d1` is `g^8`.
     // TODO(mimoo): should we instead panic/return an error if any of these return None?
-    pub fn create(n: usize) -> Option<Self> {
-        let n = Domain::<F>::compute_size_of_domain(n)?;
+    pub fn create(n: usize) -> Result<Self, ProverError> {
+        let n = Domain::<F>::compute_size_of_domain(n)
+            .ok_or(ProverError::Prover("could not compute size of domain n"))?;
 
-        let d1 = Domain::<F>::new(n)?;
+        let d1 = Domain::<F>::new(n).ok_or(ProverError::Prover(
```

Comment:
I don't think these should be `ProverError` as we're doing setup stuff here. Maybe worth creating a `SetupError`?